docs/reference/contributing/target/QuadSPI.md

```diff
@@ -1,25 +1,28 @@
 <h3 id="quadspi-port">QuadSPI</h3>
 
-[Include a brief description here.]
+QSPI HAL defines API for targets that contain QSPI capable peripheral. The QSPI interface is often used for data storage.
 
 #### Assumptions
 
 ##### Defined behavior
 
-[Include any defined behavior in bullet format here.]
+- a target implementaion should cover the most of QSPI frame format (some targets might not provide the flexibility for setting all frame parameters)
+- command transfer - a target might provide additional API for sending device specific commands. In case it does not, it can be implemented via read/write functions
 
 ##### Undefined behavior
 
-[Include any undefined behavior in bullet format here.]
+- Calling any function other than `qspi_init` before the initialization of the QSPI
 
-##### Potential bugs
+#### Dependency
 
-[Include any potential bugs in bullet format here.]
+QSPI peripheral
 
 #### Implementing QuadSPI
 
-[Include implementation information here.]
+The target needs to define `qspi_s` structure - target specific QSPI object, enable QSPI in targets.json file `device_has` and implement QSPI HAL functions defined in `hal/qspi_api.h` header file.
+
+`qspi_write` and `qspi_read` are used for data transfers. For communicating with device, `qspi_command_transfer` should be used.
 
```

Comment:
This is a good start, please add detailed documentation for each API. Also mention how the peripheral is configured, (like different operating modes, and what's supported with this implementation) and I would also like to see more info around what switches needs to be used in json files to configure this feature.

Reply:
The  API is documented in the header file, do we duplicate it here as well? I dont think so (neither can find anywhere in this contribution folder this ). I would expect the API to be documented well and we can share the pointer here 

> Also mention how the peripheral is configured, (like different operating modes, and what's supported with this implementation) and I would also like to see more info around what switches needs to be used in json files to configure this feature.

I'll add some info